### ISSUE TITLE
Disabling pod security policy is failing 

### ISSUE THREAD

**Issue by Shelby-Jiao** [terraform-google-modules/terraform-google-kubernetes-engine#1780]
### TL;DR

provider: 4.84.0
gke module: 25.0.0
source: terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster

Setting enable_pod_security_policy = true - enables admission controller and works fine.
After enabling the admission controller setting enable_pod_security_policy = fasle fails with error

`module.gke.google_container_cluster.primary: Still modifying... 

Error: googleapi: Error 400: Must specify a field to update., badRequest`

### Expected behavior

dynamic block 
<img width="699" alt="image" src="https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/assets/99865457/76539b0c-3678-4440-9455-0ccfdb56654e">
should set to `[]` when input is `false`

### Terraform Configuration

```hcl
`enable_pod_security_policy = false`
```

### Terraform Version

```sh
N/A
```

**Comment by rosie-gao-awx** [comment]
any update about this? found the same issue in my clusters
